cmd/admin-handlers.go

```diff
@@ -767,10 +745,19 @@ func (adminAPI adminAPIHandlers) HealFormatHandler(w http.ResponseWriter, r *htt
 	writeSuccessResponseHeadersOnly(w)
 }
 
-// GetConfigHandler - GET /?config
-// - x-minio-operation = get
+// GetConfigHandler - GET /minio/admin/v1/config
 // Get config.json of this minio setup.
-func (adminAPI adminAPIHandlers) GetConfigHandler(w http.ResponseWriter, r *http.Request) {
+func (adminAPI adminAPIHandlers) GetConfigHandler(w http.ResponseWriter,
+	r *http.Request) {
+
+	// Reject the request on non-TLS connection (though
+	// credentials/config have possibly been transmitted over the
+	// wire already)
+	if r.TLS == nil {
+		writeErrorResponseJSON(w, ErrAdminNonTLSCredsUpdate, r.URL)
+		return
+	}
```

Comment:
In CF and docker type deployments the TLS termination happens somewhere else hence it is better not to have this check here.

Reply:
Removed this restriction now.